### PR TITLE
Rn 899 qr code refactor

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- refactor of dashboard items to a `context` model, so that props don't get passed around as much

### CODE REVIEW

File: packages/ui-map-components/src/utils/markerColors.ts [hunk 1]
```diff
@@ -7,10 +7,7 @@
 import moment from 'moment';
 import { blue, red, green } from '@material-ui/core/colors';
 import { ScaleType, MeasureColorScheme } from '@tupaia/types';
-import {
-  BREWER_PALETTE,
-  HEATMAP_UNKNOWN_COLOR,
-} from '../constants';
+import { BREWER_PALETTE, HEATMAP_UNKNOWN_COLOR } from '../constants';
```
Comment (acdunham) [inline packages/ui-map-components/src/utils/markerColors.ts:10]: This is just formatting

File: packages/ui-map-components/package.json [hunk 2]
```diff
@@ -1,49 +1,50 @@
...
-        "@material-ui/styles": "^4.9.10",
-        "@mui/types": "^7.2.4",
-        "@storybook/react": "^6.3.9",
-        "@types/jest": "^29.5.1",
-        "@types/leaflet": "^1.7.1",
-        "@types/react-dom": "^16.9.18",
-        "@types/styled-components": "^5.1.26",
-        "react-docgen-typescript-plugin": "^1.0.5"
-    }
+  "name": "@tupaia/ui-map-components",
```
Comment (acdunham) [inline packages/ui-map-components/package.json:2]: This is formatting + fixing a type error that broke the build

File: packages/tupaia-web/src/features/DashboardItem/ExpandItemButton.tsx [hunk 3]
```diff
@@ -60,29 +60,49 @@ const ZoomInIcon = styled(MuiZoomIcon)`
...
+  };
+
+  if (!getIsExpandable()) return null;
+
+  const getText = () => {
+    if (viewType === 'dataDownload' || viewType === 'qrCodeVisual')
+      return 'Expand to download data';
+    return 'Expand chart';
+  };
+
```
Comment (rohan-bes) [inline packages/tupaia-web/src/features/DashboardItem/ExpandItemButton.tsx:94]: Glad to have this living here rather than where it was previously 👍

**Review by rohan-bes — APPROVED** [review]
Thanks for putting in the effort to explore and clean up this one @acdunham 🙏 

I really like the addition of the context to share the dashboard item state throughout the component tree. Much nicer than passing through props everywhere.

I took a stab at pushing this one a bit further to iron out the 'isExpandable' logic. I think there's potential for us to get to a really clean place, but the code diff was starting to get pretty large. Also, a tech design is probably needed to create some clearer concepts and for us to decide what the viz components should be responsible for.

Happy to take the wins you've done here and plan to take things further in the maintenance sprints.

**Pull request merged**